tests/core/test_join_resource_batches.py

```diff
@@ -0,0 +1,136 @@
+import polars as pl
+from polars.testing import assert_frame_equal
+from pytest import fixture
+
+from seedcase_sprout.core.constants import BATCH_TIMESTAMP_COLUMN_NAME
+from seedcase_sprout.core.examples import example_resource_properties
+from seedcase_sprout.core.join_resource_batches import join_resource_batches
+from seedcase_sprout.core.properties import (
+    ResourceProperties,
+)
+
+
+@fixture
+def data_list() -> list[pl.DataFrame]:
+    return [
+        pl.DataFrame(
+            {
+                "id": [0, 1],
+                "name": ["anne", "belinda"],
+                "value": [0.0, 1.1],
+                # timestamp col from `read_resource_batches`
+                BATCH_TIMESTAMP_COLUMN_NAME: ["2025-03-26T100000Z"] * 2,
+            }
+        ),
+        pl.DataFrame(
+            {
+                "id": [2, 3, 0, 0, 0],
+                "name": ["catherine", "dorothy", "anne", "anne", "alberta"],
+                "value": [2.2, 3.3, 0.0, 9.9, 0.0],
+                # timestamp col from `read_resource_batches` (different year than above)
+                BATCH_TIMESTAMP_COLUMN_NAME: ["2024-03-26T100000Z"] * 5,
+            }
+        ),
+    ]
+
+
+@fixture
+def resource_properties() -> ResourceProperties:
+    """Fixture for resource_properties."""
+    return example_resource_properties()
+
+
+def test_batches_are_joined_correctly(data_list, resource_properties):
+    """Test that the batches are joined correctly."""
+    # Given
+    resource_properties.schema.primary_key = ["id"]
```

Comment:
```suggestion
    resource_properties.schema.primary_key = "id"
```
So that we test a case where it is not a list but a string?